src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Dependencies/IDependencyCollection.cs

```diff
@@ -1,18 +1,49 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
 
 namespace Microsoft.DotNet.UpgradeAssistant.Dependencies
 {
     public interface IDependencyCollection<T> : IEnumerable<T>
     {
+        bool Add(T item, OperationDetails details);
+
+        bool Remove(T item, OperationDetails details);
+
+        [Obsolete("Use the newer Add API", error: false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         bool Add(T item, BuildBreakRisk risk = BuildBreakRisk.None);
 
+        [Obsolete("Use the newer Remove API", error: false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
         bool Remove(T item, BuildBreakRisk risk = BuildBreakRisk.None);
 
-        IReadOnlyCollection<T> Additions { get; }
+        IReadOnlyCollection<Operation<T>> Additions { get; }
```

Comment:
What about the old interface properties of Additions, and Removals? Not sure if that can be shimmed over, just wanted to know it was intentional.

Reply:
its going to be hard to shim the Additions and Deletions collection.